app/services/rent_service.rb

```diff
@@ -0,0 +1,56 @@
+module RentService
+  def self.reservation(user, rent)
+    Rent.transaction do
+      rent.validate!
+      rent.user = user
+      rent.status = user.try(:member?) ? :confirmed : :unconfirmed
+      rent.save
+      if rent.council.present?
+        rent.overlap.each(&:overbook)
+      end
+      RentMailer.rent_email(rent).deliver_now
+      true
+    end
+  end
+
+  def self.update(params, user, rent)
+    Rent.transaction do
```

Comment:
This transaction contains just one update. All updates are already in their own transaction, you don't have to write it out.